qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java

```diff
@@ -8,202 +8,154 @@
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
 import java.net.URI;
 import java.net.http.HttpClient;
 import java.net.http.HttpRequest;
 import java.net.http.HttpRequest.BodyPublishers;
-import java.net.http.HttpResponse.BodyHandlers;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandler;
+import java.net.http.HttpResponse.BodySubscriber;
+import java.net.http.HttpResponse.ResponseInfo;
 import java.time.Duration;
 import java.time.Instant;
-import java.util.Collections;
-import java.util.List;
-import java.util.Objects;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.stream.Collectors;
-import org.agrona.CloseHelper;
-import org.awaitility.Awaitility;
-import org.elasticsearch.client.RestClient;
-import org.junit.jupiter.api.AfterEach;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.testcontainers.containers.Network;
-import org.testcontainers.elasticsearch.ElasticsearchContainer;
-import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
 
+// NOTE: once https://github.com/camunda-community-hub/zeebe-test-container/issues/318 is completed,
+// we can reduce execution time by sharing the same engine and resetting the clock in between
+@Testcontainers
 final class ControlledActorClockEndpointIT {
-  private static final String ELASTICSEARCH_HOST = "elasticsearch";
-  private static final String INDEX_PREFIX = "exporter-clock-test";
-  private Network network;
-  private ElasticsearchContainer elasticsearchContainer;
-  private ZeebeContainer zeebeContainer;
-  private ZeebeClient zeebeClient;
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper().registerModule(new JavaTimeModule());
+
+  private final ZeebeContainer container =
+      new ZeebeContainer(ZeebeTestContainerDefaults.defaultTestImage())
+          .withEnv("ZEEBE_CLOCK_CONTROLLED", "true");
+
+  @Container
+  private final ContainerEngine engine =
+      ContainerEngine.builder()
+          .withContainer(container)
+          .withIdlePeriod(Duration.ofSeconds(2))
+          .build();
+
   private final HttpClient httpClient = HttpClient.newHttpClient();
-  private final ObjectMapper mapper = new ObjectMapper().registerModule(new ZeebeProtocolModule());
-  private ZeebeClock zeebeClock;
+  private ZeebeClient zeebeClient;
 
   @BeforeEach
-  void startContainers() {
-    network = Network.newNetwork();
-    startElasticsearch();
-    startZeebe();
-  }
-
-  @AfterEach
-  void stopContainers() {
-    CloseHelper.closeAll(zeebeClient, zeebeContainer, elasticsearchContainer, network);
+  void beforeEach() {
+    zeebeClient = engine.createClient();
```

Comment:
FYI, the `ContainerEngine` extension will take care of cleaning up any clients when it closes.